{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "import PyPDF2 as pdf\n",
    "from tabula import read_pdf\n",
    "from tabulate import tabulate\n",
    "import pandas as pd\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "using the following pdf files:\n",
      "['CCGL_3152_FinalReport_Sandler_adult_GBM_IDH1_R132H_ATRX_CDKN2A__no_TP53_.pdf', 'CCGL_3149_FinalReport_adult_recurrent_astrocytoma_IDH1_R132H_TP53_ATRX_CDKN2A_PDGFRA_amp.pdf', 'CCGL_2831_FinalReport_adult_astrocytoma_IDH1_R132H_TP53_ATRX.pdf', 'CCGL_2835_FinalReport_adult_astrocytoma_IDH1_R132H_ATRX_TP53_PIK3R1.pdf', 'CCGL_3205_FinalReport_pedi_LFS_associated_diffuse_astrocytoma_IDH1_R132H_germline_TP53.pdf', 'CCGL_3619_FinalReport_adult_Ollier_associated_astrocytoma_IDH1_R132H_TP53_ATRX_NF1_SPRED1_NOTCH1_constitutional_mosaicism_IDH1.pdf']\n",
      "Patient Name: CCGL-3152\n",
      "Transcript Ids being retrieved: ['NM_000489', 'all', 'NM_005896']\n",
      "Reading from CCGL-3152_CGP-9473_T_signout.xlsx\n",
      "table made\n",
      "Patient Name: CCGL-3149\n",
      "Transcript Ids being retrieved: ['NM_000489', 'all', 'NM_005896', 'all', 'NM_000546']\n",
      "Reading from CCGL-3149_CGP-9470_T_signout.xlsx\n",
      "table made\n",
      "Patient Name: CCGL-2831\n",
      "Transcript Ids being retrieved: ['NM_000489', 'NM_005896', 'NM_000546']\n",
      "Reading from CCGL-2831_CGP-8784_T_signout.xlsx\n",
      "table made\n",
      "Patient Name: CCGL-2835\n",
      "Transcript Ids being retrieved: ['NM_000489', 'NM_005896', 'NM_181523', 'NM_000546']\n",
      "Reading from CCGL-2835_CGP-8785_T_signout.xlsx\n",
      "table made\n",
      "Patient Name: CCGL-3205\n",
      "Transcript Ids being retrieved: ['NM_005896', 'NM_000546']\n",
      "Reading from CCGL-3205%2CCCGL-1628_CGP-9586_T_CGP-6259_N_signout.xlsx\n",
      "table made\n",
      "Patient Name: CCGL-3619\n",
      "Transcript Ids being retrieved: ['NM_000489', 'NM_000489', 'NM_000489', 'NM_005896', 'NM_001042492', 'NM_017617', 'NM_152594', 'NM_000546']\n",
      "Reading from CCGL-3619_CGP-10495_T_CGP-10511_N_signout.xlsx\n",
      "table made\n"
     ]
    }
   ],
   "source": [
    "def gener_gene_data(pdf_path):\n",
    "    pdf = read_pdf(pdf_path, pages=1)\n",
    "    pd_df = pdf[1]\n",
    "    pd_df = pd_df.rename(columns=pd_df.iloc[0]).drop(pd_df.index[0])\n",
    "\n",
    "    id_list = list(pd_df[\"TRANSCRIPT ID\"])\n",
    "    id_list = [x.split('.')[0] for x in id_list]\n",
    "    patient_name = '-'.join(pdf_path.split('_')[:2])\n",
    "    \n",
    "    print(\"Patient Name: \" + patient_name)\n",
    "    print(\"Transcript Ids being retrieved: \" + str(id_list))\n",
    "    \n",
    "    xl_files_og = [f for f in os.listdir('.') if os.path.isfile(f) and f.endswith('.xlsx')]\n",
    "    xl_files = [f.replace('%', '_') for f in xl_files_og]\n",
    "    xl_files = [f.split('_')[0] for f in xl_files]\n",
    "    ind = 0\n",
    "    for i in range(0, len(xl_files)):\n",
    "        if patient_name == xl_files[i]:\n",
    "            ind = i\n",
    "    \n",
    "\n",
    "    print(\"Reading from \" + xl_files_og[ind])\n",
    "    gene_data = pd.read_excel(xl_files_og[ind])\n",
    "    # gene_data[\"Variant Filter\"] = gene_data.apply(lambda s: s.Variant.split(' ')[0], axis=1)\n",
    "    gd = gene_data.loc[gene_data[\"TranscriptID\"].isin(id_list)]\n",
    "    print(\"table made\")\n",
    "    return gd  \n",
    "\n",
    "# gener_gene_data(\"CCGL_3205_FinalReport_pedi_LFS_associated_diffuse_astrocytoma_IDH1_R132H_germline_TP53.pdf\")\n",
    "\n",
    "files = [f for f in os.listdir('.') if os.path.isfile(f) and f.endswith('.pdf')]\n",
    "print(\"using the following pdf files:\")\n",
    "print(files)\n",
    "data_reports = {f: gener_gene_data(f) for f in files}\n",
    "# data_reports\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Func</th>\n",
       "      <th>T:FUP</th>\n",
       "      <th>Variant</th>\n",
       "      <th>TranscriptID</th>\n",
       "      <th>Classification</th>\n",
       "      <th>T:COV</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomADexome</th>\n",
       "      <th>esp6500si</th>\n",
       "      <th>1000g2015aug</th>\n",
       "      <th>snp150</th>\n",
       "      <th>ucsf500normT</th>\n",
       "      <th>ucsf500normN</th>\n",
       "      <th>cosmic87</th>\n",
       "      <th>cbioportal2018nov</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>CADD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>chrX</td>\n",
       "      <td>76937963</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>110</td>\n",
       "      <td>ATRX p.Q929E</td>\n",
       "      <td>NM_000489</td>\n",
       "      <td>NaN</td>\n",
       "      <td>415</td>\n",
       "      <td>...</td>\n",
       "      <td>37.3300%</td>\n",
       "      <td>50.8300%</td>\n",
       "      <td>55.2318%</td>\n",
       "      <td>rs3088074</td>\n",
       "      <td>34.5560%</td>\n",
       "      <td>34.5560%</td>\n",
       "      <td>ID=COSM3759543,COSM3759544;OCCURRENCE=2(breast...</td>\n",
       "      <td>4:ATRX:Q929E(4)</td>\n",
       "      <td>CLINSIG=Benign;CLNDN=not_specified;CLNREVSTAT=...</td>\n",
       "      <td>0.391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7579472</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>110</td>\n",
       "      <td>TP53 p.P72R</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>409</td>\n",
       "      <td>...</td>\n",
       "      <td>66.8600%</td>\n",
       "      <td>63.0000%</td>\n",
       "      <td>54.2931%</td>\n",
       "      <td>rs1042522</td>\n",
       "      <td>69.3880%</td>\n",
       "      <td>69.3610%</td>\n",
       "      <td>ID=COSM250061,COSM3766193,COSM3766192,COSM3766...</td>\n",
       "      <td>33:TP53:P72R(33)</td>\n",
       "      <td>CLINSIG=drug_response;CLNDN=CODON_72_POLYMORPH...</td>\n",
       "      <td>8.316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7578407</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>110</td>\n",
       "      <td>TP53 p.R175G</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>304</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs138729528</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ID=COSM4070040,COSM707882,COSM707883,COSM16408...</td>\n",
       "      <td>21:TP53:R175G(21)</td>\n",
       "      <td>CLINSIG=Pathogenic/Likely_pathogenic;CLNDN=Med...</td>\n",
       "      <td>18.190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>chr5</td>\n",
       "      <td>67591104</td>\n",
       "      <td>TTAA</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>101</td>\n",
       "      <td>PIK3R1 p.566_567del</td>\n",
       "      <td>NM_181523</td>\n",
       "      <td>NaN</td>\n",
       "      <td>752</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ID=COSM5835073,COSM5835072,COSM5835074;OCCURRE...</td>\n",
       "      <td>3:PIK3R1:K567del(3)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>76939929</td>\n",
       "      <td>C</td>\n",
       "      <td>CCT</td>\n",
       "      <td>exonic</td>\n",
       "      <td>111</td>\n",
       "      <td>ATRX p.L273fs</td>\n",
       "      <td>NM_000489</td>\n",
       "      <td>NaN</td>\n",
       "      <td>375</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS   REF  ALT    Func  T:FUP              Variant  \\\n",
       "67    chrX  76937963     G    C  exonic    110         ATRX p.Q929E   \n",
       "69   chr17   7579472     G    C  exonic    110          TP53 p.P72R   \n",
       "78   chr17   7578407     G    C  exonic    110         TP53 p.R175G   \n",
       "105   chr5  67591104  TTAA    T  exonic    101  PIK3R1 p.566_567del   \n",
       "202   chrX  76939929     C  CCT  exonic    111        ATRX p.L273fs   \n",
       "\n",
       "    TranscriptID  Classification  T:COV  ... gnomADexome  esp6500si  \\\n",
       "67     NM_000489             NaN    415  ...    37.3300%   50.8300%   \n",
       "69     NM_000546             NaN    409  ...    66.8600%   63.0000%   \n",
       "78     NM_000546             NaN    304  ...         NaN        NaN   \n",
       "105    NM_181523             NaN    752  ...         NaN        NaN   \n",
       "202    NM_000489             NaN    375  ...         NaN        NaN   \n",
       "\n",
       "    1000g2015aug       snp150 ucsf500normT ucsf500normN  \\\n",
       "67      55.2318%    rs3088074     34.5560%     34.5560%   \n",
       "69      54.2931%    rs1042522     69.3880%     69.3610%   \n",
       "78           NaN  rs138729528          NaN          NaN   \n",
       "105          NaN          NaN          NaN          NaN   \n",
       "202          NaN          NaN          NaN          NaN   \n",
       "\n",
       "                                              cosmic87    cbioportal2018nov  \\\n",
       "67   ID=COSM3759543,COSM3759544;OCCURRENCE=2(breast...      4:ATRX:Q929E(4)   \n",
       "69   ID=COSM250061,COSM3766193,COSM3766192,COSM3766...     33:TP53:P72R(33)   \n",
       "78   ID=COSM4070040,COSM707882,COSM707883,COSM16408...    21:TP53:R175G(21)   \n",
       "105  ID=COSM5835073,COSM5835072,COSM5835074;OCCURRE...  3:PIK3R1:K567del(3)   \n",
       "202                                                NaN                  NaN   \n",
       "\n",
       "                                               clinvar    CADD  \n",
       "67   CLINSIG=Benign;CLNDN=not_specified;CLNREVSTAT=...   0.391  \n",
       "69   CLINSIG=drug_response;CLNDN=CODON_72_POLYMORPH...   8.316  \n",
       "78   CLINSIG=Pathogenic/Likely_pathogenic;CLNDN=Med...  18.190  \n",
       "105                                                NaN     NaN  \n",
       "202                                                NaN     NaN  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = data_reports.get(\"CCGL_2835_FinalReport_adult_astrocytoma_IDH1_R132H_ATRX_TP53_PIK3R1.pdf\")\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdfFile = open(\"CCGL_2831_FinalReport_adult_astrocytoma_IDH1_R132H_TP53_ATRX.pdf\", \"rb\")\n",
    "pdf = read_pdf(\"CCGL_2831_FinalReport_adult_astrocytoma_IDH1_R132H_TP53_ATRX.pdf\", pages=\"all\")\n",
    "pd_df = pdf[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd_df = pd_df.rename(columns=pd_df.iloc[0]).drop(pd_df.index[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT</th>\n",
       "      <th>TRANSCRIPT ID</th>\n",
       "      <th>CLASSIFICATION</th>\n",
       "      <th>READS</th>\n",
       "      <th>MUTANT ALLELE\\rFREQUENCY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATRX p.M1679fs</td>\n",
       "      <td>NM_000489.3</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>379</td>\n",
       "      <td>51%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH1 p.R132H</td>\n",
       "      <td>NM_005896.2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>780</td>\n",
       "      <td>25%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53 p.R273C</td>\n",
       "      <td>NM_000546.5</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>334</td>\n",
       "      <td>54%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          VARIANT TRANSCRIPT ID CLASSIFICATION READS MUTANT ALLELE\\rFREQUENCY\n",
       "1  ATRX p.M1679fs   NM_000489.3     Pathogenic   379                      51%\n",
       "2    IDH1 p.R132H   NM_005896.2     Pathogenic   780                      25%\n",
       "3    TP53 p.R273C   NM_000546.5     Pathogenic   334                      54%"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ATRX', 'IDH1', 'TP53']"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_list = [x.split(' ')[0] for x in list(pd_df[\"VARIANT\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NM_000489', 'NM_005896', 'NM_000546']"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_list = list(pd_df[\"TRANSCRIPT ID\"])\n",
    "id_list = [x.split('.')[0] for x in id_list]\n",
    "id_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Func</th>\n",
       "      <th>T:FUP</th>\n",
       "      <th>Variant</th>\n",
       "      <th>TranscriptID</th>\n",
       "      <th>Classification</th>\n",
       "      <th>T:COV</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomADexome</th>\n",
       "      <th>esp6500si</th>\n",
       "      <th>1000g2015aug</th>\n",
       "      <th>snp150</th>\n",
       "      <th>ucsf500normT</th>\n",
       "      <th>ucsf500normN</th>\n",
       "      <th>cosmic87</th>\n",
       "      <th>cbioportal2018nov</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>CADD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7579472</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>110</td>\n",
       "      <td>TP53 p.P72R</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>336</td>\n",
       "      <td>...</td>\n",
       "      <td>66.8600%</td>\n",
       "      <td>63.0000%</td>\n",
       "      <td>54.2931%</td>\n",
       "      <td>rs1042522</td>\n",
       "      <td>69.3880%</td>\n",
       "      <td>69.3610%</td>\n",
       "      <td>ID=COSM250061,COSM3766193,COSM3766192,COSM3766...</td>\n",
       "      <td>33:TP53:P72R(33)</td>\n",
       "      <td>CLINSIG=drug_response;CLNDN=CODON_72_POLYMORPH...</td>\n",
       "      <td>8.316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7577121</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>110</td>\n",
       "      <td>TP53 p.R273C</td>\n",
       "      <td>NM_000546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>334</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0012%</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs121913343</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ID=COSM3355991,COSM99933,COSM10659,COSM1645518...</td>\n",
       "      <td>429:TP53:R273C(429)</td>\n",
       "      <td>CLINSIG=Conflicting_interpretations_of_pathoge...</td>\n",
       "      <td>17.480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>chrX</td>\n",
       "      <td>76888758</td>\n",
       "      <td>GAGCAAGATTTCTATACATCTCATAGCCTATGATCATA</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>101</td>\n",
       "      <td>ATRX p.V1678fs</td>\n",
       "      <td>NM_000489</td>\n",
       "      <td>NaN</td>\n",
       "      <td>379</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM       POS                                     REF ALT    Func  \\\n",
       "63   chr17   7579472                                       G   C  exonic   \n",
       "64   chr17   7577121                                       G   A  exonic   \n",
       "182   chrX  76888758  GAGCAAGATTTCTATACATCTCATAGCCTATGATCATA   G  exonic   \n",
       "\n",
       "     T:FUP         Variant TranscriptID  Classification  T:COV  ...  \\\n",
       "63     110     TP53 p.P72R    NM_000546             NaN    336  ...   \n",
       "64     110    TP53 p.R273C    NM_000546             NaN    334  ...   \n",
       "182    101  ATRX p.V1678fs    NM_000489             NaN    379  ...   \n",
       "\n",
       "    gnomADexome  esp6500si 1000g2015aug       snp150 ucsf500normT  \\\n",
       "63     66.8600%   63.0000%     54.2931%    rs1042522     69.3880%   \n",
       "64      0.0012%        NaN          NaN  rs121913343          NaN   \n",
       "182         NaN        NaN          NaN          NaN          NaN   \n",
       "\n",
       "    ucsf500normN                                           cosmic87  \\\n",
       "63      69.3610%  ID=COSM250061,COSM3766193,COSM3766192,COSM3766...   \n",
       "64           NaN  ID=COSM3355991,COSM99933,COSM10659,COSM1645518...   \n",
       "182          NaN                                                NaN   \n",
       "\n",
       "       cbioportal2018nov                                            clinvar  \\\n",
       "63      33:TP53:P72R(33)  CLINSIG=drug_response;CLNDN=CODON_72_POLYMORPH...   \n",
       "64   429:TP53:R273C(429)  CLINSIG=Conflicting_interpretations_of_pathoge...   \n",
       "182                  NaN                                                NaN   \n",
       "\n",
       "       CADD  \n",
       "63    8.316  \n",
       "64   17.480  \n",
       "182     NaN  \n",
       "\n",
       "[3 rows x 26 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_data = pd.read_excel(\"CCGL-2831_CGP-8784_T_signout.xlsx\")\n",
    "# gene_data[\"Variant Filter\"] = gene_data.apply(lambda s: s.Variant.split(' ')[0], axis=1)\n",
    "gd = gene_data.loc[gene_data[\"TranscriptID\"].isin(id_list)]\n",
    "gd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdfReader = pdf.PdfReader(pdfFile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<PyPDF2._reader.PdfReader object at 0x7fdf90acb4c0>\n"
     ]
    }
   ],
   "source": [
    "print(pdfReader)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: tabulate in /Users/yajatyadav/opt/anaconda3/lib/python3.9/site-packages (0.8.10)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install tabulate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "73426f12d1759081d99caedc0e112dcda7a31d2029c90e0b4f139d4ebd81d65c"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
